sdl2promela/stop_condition/translator.py

```diff
@@ -278,36 +291,165 @@ def _generate(context: GenerateContext, expr: model.VariableReference):
     """
     if context.choice_selection is not None:
+        # The list of available alternatives of CHOICE is case-sensitive
+        # Stop Condition Language is case-insensitive,
+        # so, the existing identifier should be generated.
+        selected_alternatives = [
+            alternative
+            for alternative in context.choice_selection_alternatives
+            if alternative.lower() == expr.name.lower()
+        ]
+        if len(selected_alternatives) != 1:
+            raise TranslateException(
+                f"The CHOICE datatype '{context.choice_selection}' has no alternative with name '{expr.name}'"
+            )
+
         return promelaBuilder.VariableReferenceBuilder(
-            context.choice_selection + "_" + expr.name + "_PRESENT"
+            context.choice_selection + "_" + selected_alternatives[0] + "_PRESENT"
         ).build()
     else:
         return promelaBuilder.VariableReferenceBuilder(expr.name).build()
 
 
+class FirstVariableInfo:
+    """
+    Information about first variable of selector.
+    """
+
+    process_name: str
+    """Name of process."""
+
+    variable_name: str
+    """Name of first variable."""
+
+    variable_type: Any
+    """Type of variable from OpenGEODE."""
+
+    consumed_elements: int
+    """Number of consumed elements of Selector."""
+
+    def __init__(self, process_name, variable_name, variable_type, consumed_elements):
+        self.process_name = process_name
+        self.variable_name = variable_name
+        self.variable_type = variable_type
+        self.consumed_elements = consumed_elements
+
+
+def _find_first_variable(
+    context: GenerateContext, selector: model.Selector
+) -> FirstVariableInfo:
+    """
+    Find first variable referenced by selector.
+    """
+    index = 0
+
+    # Find top-level process
+    process_name = typing.cast(
+        model.VariableReference, selector.elements[index]
+    ).name.lower()
+    if process_name not in context.processes:
+        raise TranslateException(
+            "Cannot find process with name '{}'".format(process_name)
+        )
+
+    process = context.processes[process_name]
+    aggregates = context.aggregates[process_name]
+    index = index + 1
+    composites = process.composite_states
+
+    variable_prefix = ""
+    state_name = ""
+
+    # Check if composite (nested) or aggregate (parallel) states exist
+    # And construct variable prefix, which consist of state name.
+    while True:
+        element_name = typing.cast(
+            model.VariableReference, selector.elements[index]
+        ).name.lower()
+        if len(state_name) == 0:
+            state_name = element_name
+        else:
+            state_name = state_name + SEPARATOR + element_name
+        candidates = [
+            composite for composite in composites if composite.statename == state_name
+        ]
+        if len(candidates) > 0:
+            process = candidates[0]
```

Comment:
What if len > 1?

Reply:
It shouldn't be possible, this means that opengeode has two different states with same name...